src/napari_spatialdata/_viewer.py

```diff
@@ -403,24 +424,47 @@ def add_sdata_points(self, sdata: SpatialData, key: str, selected_cs: str, multi
             gen = np.random.default_rng()
             subsample = np.sort(gen.choice(len(points), size=100000, replace=False))  # same as indices
 
+        # TODO consider subsampling adata and passing that on.
+        adata, table_name, table_names = self._get_table_data(sdata, original_name)
```

Comment:
Here the function `_get_table_data` returns `adata`, but this is not passed to `add_points`. This could be a double problem:
1. the adata (and the eventual join) performed in `_get_table_data()`, could be expensive, and they are not used
2. the subset operation that is done below in-line in `add_points()` is not a full join, so in the case in which the table mismatches the points it would lead to errors.

I would address this here by replacing the call of `_get_table_data(sdata)` with `get_table_data(...)` where the argument is a subset (=copied) of the points. If this is done I think this will address the TODO, so it can be removed.